src/schema.ts

```diff
@@ -319,58 +296,37 @@ export interface VibraphoneVibratoSpeedEvent {
 }
 
 /**
- * Represents a single string on the bass.
- *
- * We chose to start at one since musicians count
- * starting from one. String 1 is highest pitched
- * string, and string 4 is the lowest pitched string.
+ * Any event pertaining to the vibraphone.
  */
-export type BassString = 1 | 2 | 3 | 4;
-
-/** Represents the state pertaining to the bass. */
-export interface BassState {
-	/**
-	 * Which frets the capos are on.
-	 * `0` or nothing means no capo is applied to that string.
-	 */
-	capos: { [S in BassString]?: number };
-	/**
-	 * Which notes the strings are tuned to.
-	 * Nothing means the standard bass tuning:
-	 * ```typescript
-	 * {
-	 *   4: "E1",
-	 *   3: "A1",
-	 *   2: "D2",
-	 *   1: "G2",
-	 * }
-	 * ```
-	 */
-	tuning: { [S in BassString]?: Note };
-}
+export type VibraphoneEvent =
+	| VibraphoneVibratoEnabledEvent
+	| VibraphoneVibratoSpeedEvent;
 
-/** Any event pertaining to the bass. */
-export type BassEvent = BassCapoEvent;
 /** An event corresponding to a capo being applied, moved, or removed. */
-export interface BassCapoEvent {
-	kind: "bass_capo";
+export interface CapoFretEvent {
+	kind: "capo_fret";
 	/** Which string is affected. */
-	capoString: BassString;
+	bassString: BassString;
 	/**
 	 * Which fret is the capo to be applied to.
 	 * `0` or nothing means no capo is applied.
 	 */
 	fret: number;
 }
 
-/** Represents the state pertaining to the hihat machine. */
-export interface HihatMachineState {
-	/** The meaning of this property has yet to be determined. */
-	setting: string;
+export interface CapoEngagedEvent {
+	kind: "capo_engaged";
+	/** Which string is affected. */
+	bassString: BassString;
+	/**
+	 * If the capo is engaged
+	 */
+	engaged: boolean;
```

Comment:
A fret of 0 would be the same as disengaged.

Reply:
The actual machine will have capos that can be positioned at a non 0 position but be disengaged, this again is to match the physical behaviour of the machine.

Reply:
Ok.